smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.typescript.codegen;
+
+import software.amazon.smithy.build.FileManifest;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.TopDownIndex;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+
+/**
+ * Generates an index to export the service client and each command.
+ */
+final class IndexGenerator {
+
+    private IndexGenerator() {}
+
+    static void writeIndex(
+        TypeScriptSettings settings,
+        Model model,
+        SymbolProvider symbolProvider,
+        FileManifest fileManifest
+    ) {
+        TypeScriptWriter writer = new TypeScriptWriter("");
+        ServiceShape service = settings.getService(model);
+        Symbol symbol = symbolProvider.toSymbol(service);
+        writer.write("export * from \"./" + symbol.getName() + "\";");
```

Comment:
Does this need to export the [non-modular client](https://github.com/awslabs/smithy-typescript/blob/master/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java#L232)?

Reply:
Added in https://github.com/awslabs/smithy-typescript/pull/34/commits/6046592d5d10cc36f6420ba3faab29bed072578a